src/adapter/src/coord/ddl.rs

```diff
@@ -321,62 +338,47 @@ impl<S: Append + 'static> Coordinator<S> {
         oid: u32,
-        connection: SinkConnection<()>,
-        compute_instance: ComputeInstanceId,
+        connection: StorageSinkConnection,
         session: Option<&Session>,
     ) -> Result<(), AdapterError> {
         // Update catalog entry with sink connection.
         let entry = self.catalog.get_entry(&id);
         let name = entry.name().clone();
-        let mut sink = match entry.item() {
-            CatalogItem::Sink(sink) => sink.clone(),
+        let sink = match entry.item() {
+            CatalogItem::Sink(sink) => sink,
             _ => unreachable!(),
         };
-        sink.connection = catalog::SinkConnectionState::Ready(connection.clone());
-        // We don't try to linearize the as of for the sink; we just pick the
-        // least valid read timestamp. If users want linearizability across
-        // Materialize and their sink, they'll need to reason about the
-        // timestamps we emit anyway, so might as emit as much historical detail
-        // as we possibly can.
-        let id_bundle = self
-            .index_oracle(compute_instance)
-            .sufficient_collections(&[sink.from]);
-        let frontier = self.least_valid_read(&id_bundle);
-        let as_of = SinkAsOf {
-            frontier,
-            strict: !sink.with_snapshot,
+        let sink = catalog::Sink {
+            connection: StorageSinkConnectionState::Ready(connection.clone()),
+            ..sink.clone()
         };
+
         let ops = vec![
             catalog::Op::DropItem(id),
             catalog::Op::CreateItem {
                 id,
                 oid,
-                name: name.clone(),
+                name,
                 item: CatalogItem::Sink(sink.clone()),
             },
         ];
-        let df = self
-            .catalog_transact(session, ops, |txn| {
-                let mut builder = txn.dataflow_builder(compute_instance);
-                let from_entry = builder.catalog.get_entry(&sink.from);
-                let sink_description = mz_storage::types::sinks::SinkDesc {
-                    from: sink.from,
-                    from_desc: from_entry
-                        .desc(
-                            &builder
-                                .catalog
-                                .resolve_full_name(from_entry.name(), from_entry.conn_id()),
-                        )
-                        .unwrap()
-                        .into_owned(),
-                    connection: connection.clone(),
-                    envelope: Some(sink.envelope),
-                    as_of,
-                };
-                Ok(builder.build_sink_dataflow(name.to_string(), id, sink_description)?)
-            })
-            .await?;
 
-        Ok(self.ship_dataflow(df, compute_instance).await)
+        let () = self.catalog_transact(session, ops, move |_| Ok(())).await?;
+        // XXX(chae): should this happen in the same task where we build the connection??  Or should it need to happen
```

Comment:
You mean: right now it's not happening in the same task but you're wondering if it should? Before, if the sink connection "was ready" nothing could fail, right? Rendering the sink into the compute dataflow was infallible but `create_storage_export` is not.

Have you tried putting `create_storage_export()` in that same task? Is it even feasible with borrowing `self` and whatnot (just spitballing here). It does seem like it could be correct to to have it in the same task but I'm also willing to file that as a potential future TODO so that we can merge this PR.

Reply:
> Before, if the sink connection "was ready" nothing could fail, right?
well that's what we were asserting! (which was maybe not ideal -- but that's a separate issue imo)

I don't think we can do it in a straightforward way for the reason you mentioned.  I think I'll leave this as a TODO -- because I agree that I think it would be correct to have in the same task!

Reply:
Opened https://github.com/MaterializeInc/materialize/issues/14220